system/jlib/jatomic.hpp

```diff
@@ -40,6 +42,50 @@ inline static void spinPause() { } // MORE: Is there an equivalent?
 inline static void spinPause() { }
 #endif
 
+//std::atomic doesn't have an equivalent of add_fetch which takes a memory order operand...  Hide the different implementations.
+#if defined(_ARCH_X86_64_) || defined(_ARCH_X86_)
+//For x86 use the operator += since sequential consistency has no penalty
+template <typename T>
+auto add_fetch(T & value, decltype(value.load()) delta, std::memory_order order = std::memory_order_seq_cst) -> decltype(value.load()) { return value.operator += (delta); }
```

Comment:
A very good question..  I should have checked.

  16037 1ed2 F00FC110 	lock xaddl	%edx, (%rax)
  16042 1ed6 428D0422 		leal	(%rdx,%r12), %eax

v

  16029 1ed2 F00FC102 	lock xaddl	%eax, (%rdx)
  16030 1ed6 4401E0   		addl	%r12d, %eax
 
I will delete the special case code - it isn't worth including.